ethereum/sidecar/attestation_validator.go

```diff
@@ -0,0 +1,168 @@
+package sidecar
+
+import (
+	"context"
+	"fmt"
+	"math/big"
+	"time"
+
+	"cosmossdk.io/log"
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/crypto"
+	ethconfig "github.com/keep-network/keep-common/pkg/chain/ethereum"
+	ethconnect "github.com/mezo-org/mezod/ethereum"
+	"github.com/mezo-org/mezod/ethereum/bindings/portal"
+)
+
+var defaultIsValidTickerDuration = time.Second
+
+type attestationValidator struct {
+	logger         log.Logger
+	bridgeContract ethconnect.BridgeContract
+	address        common.Address
+}
+
+func newAttestationValidation(
+	logger log.Logger,
+	bridgeContract ethconnect.BridgeContract,
+	validatorAddress common.Address,
+) *attestationValidator {
+	return &attestationValidator{
+		logger:         logger,
+		bridgeContract: bridgeContract,
+		address:        validatorAddress,
+	}
+}
+
+func (av *attestationValidator) IsValid(ctx context.Context, bridgeAssetsUnlocked *portal.MezoBridgeAssetsUnlocked) (bool, error) {
+	ticker := time.NewTicker(defaultIsValidTickerDuration)
+	defer ticker.Stop()
+
+	for {
+		select {
+		case <-ticker.C:
+			ok, err := av.bridgeContract.ValidateAssetsUnlocked(*bridgeAssetsUnlocked)
+			if err != nil {
+				av.logger.Error("couldn't call validateAssetsUnlocked", "error", err)
+				continue
+			}
+			return ok, nil
+		case <-ctx.Done():
+			av.logger.Info("stopping assets unlocked validation due to context cancellation")
+			return false, ctx.Err()
+		}
+	}
```

Comment:
What's the point of the ticker here? This is a one-off action.

Reply:
0bde10d2